### DISCUSSION TITLE
How to run some extension via HTTP GET query?

### DISCUSSION THREAD

**Discussion by rei-koshka** [coder/code-server#2053] [Q&A] [answered]
Hi there!
I'd like to run any extension as it's can be done by CTRL+SHIFT+P in VS Code.
1. Is there any ready-to-use solution to do it via query parameter?
2. How can I extend current query handlers to approach this behaviour (need a little bit guidelines)?

**Comment by code-asher** [comment, marked as answer]
`CTRL+SHIFT+P` should work in code-server, is it not working for you? You might have to re-bind if you're using Firefox since that keybinding opens a private window and can't be overidden.

There isn't any way currently to open the command palette or run commands via a query parameter.

I don't think there are any existing query handlers that could be extended to provide this behavior either. Right now we only look at query variables on the initial page load so the entire page would have to be reloaded which I think would provide a pretty poor experience.

**Reply by rei-koshka** [reply]
Thanks for the reply!
Currently I don't need in shortcut by itself.
My goal is a feature of navigation to a specific line or symbol by clicking on URL. Somehow I don't care about perfomance on page reloading.  I'd try to call VS Code API to find a symbol specified in a query. Is that possible?

**Reply by code-asher** [reply]
Hmmm I don't think the extension API supports doing anything based on query variables (https://code.visualstudio.com/api/references/vscode-api) but I'm not 100% familiar with it so I might have missed something.

Depending on what kind of navigation you need you might want to check out some of these methods from the API:
- `openExternal`
- `openTextDocument`
- `executeCommand` (you can execute things like `vscode.openFolder` which would let you open a new folder)

If you absolutely need to do something with query variables I think currently the only way might be to patch VS Code. We do this to handle the `folder` query variable, for example.

**Reply by code-asher** [reply]
Oh we do write the query variables for the last loaded session to a file on disk that you could use though. It's in `[data directory]/coder.json`. So your extension could read that file to get the most recent query variables.

**Reply by rei-koshka** [reply]
Thanks! These references helped me a lot!
>executeCommand (you can execute things like `vscode.openFolder` which would let you open a new folder)

That's what I am a looking for.
